File: CS/WebApplication2/CalendarApiHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Google.Apis.Calendar.v3.Data;
using DevExpress.XtraScheduler.iCalendar.Native;

public static class CalendarApiHelper {
    public static DateTime ConvertDateTime(EventDateTime start) {
        if (start == null)
            return DateTime.MinValue;
        if (start.DateTime.HasValue)
            return start.DateTime.Value;
        return DateTime.Parse(start.Date);
    }
    public static bool IsAllDay(EventDateTime start) {
        if (start == null)
            return false;
        return !start.DateTime.HasValue;
    }
    public static string GetRawDate(DateTime date) {
        return date.ToString("yyyy-MM-dd");
    }
    public static void MakeAllDay(EventDateTime eventDate) {
        DateTime dateTime = eventDate.DateTime.Value;
        eventDate.DateTime = null;
        eventDate.DateTimeRaw = CalendarApiHelper.GetRawDate(dateTime.Date);
    }
  
    public static string GetVTimeZone(string timeZoneId)
    {
        return TimeZoneConverter.ConvertToVTimeZone(TimeZoneInfo.FindSystemTimeZoneById(timeZoneId)).TimeZoneIdentifier.Value;
    }
    internal static void MakeDate(EventDateTime eventDate) {
        DateTime dateTime = ConvertDateTime(eventDate);
        eventDate.DateTime = dateTime;
    }
}

File: VB/WebApplication2/Helpers/CalendarApiHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports Google.Apis.Calendar.v3.Data
Imports DevExpress.XtraScheduler.iCalendar.Native

Public NotInheritable Class CalendarApiHelper

    Private Sub New()
    End Sub

    Public Shared Function ConvertDateTime(ByVal start As EventDateTime) As Date
        If start Is Nothing Then
            Return Date.MinValue
        End If
        If start.DateTime.HasValue Then
            Return start.DateTime.Value
        End If
        Return Date.Parse(start.Date)
    End Function
    Public Shared Function IsAllDay(ByVal start As EventDateTime) As Boolean
        If start Is Nothing Then
            Return False
        End If
        Return Not start.DateTime.HasValue
    End Function
    Public Shared Function GetRawDate(ByVal [date] As Date) As String
        Return [date].ToString("yyyy-MM-dd")
    End Function
    Public Shared Sub MakeAllDay(ByVal eventDate As EventDateTime)
        Dim dateTime As Date = eventDate.DateTime.Value
        eventDate.DateTime = Nothing
        eventDate.DateTimeRaw = CalendarApiHelper.GetRawDate(dateTime.Date)
    End Sub

    Public Shared Function GetVTimeZone(ByVal timeZoneId As String) As String
        Return TimeZoneConverter.ConvertToVTimeZone(TimeZoneInfo.FindSystemTimeZoneById(timeZoneId)).TimeZoneIdentifier.Value
    End Function
    Friend Shared Sub MakeDate(ByVal eventDate As EventDateTime)
        Dim dateTime As Date = ConvertDateTime(eventDate)
        eventDate.DateTime = dateTime
    End Sub
End Class